coffee_lib/src/repository.rs

```diff
@@ -33,5 +33,14 @@ pub trait Repository: Any {
     /// return the url of the repository.
     fn url(&self) -> URL;
 
+    /// return the git head of the repository.
+    fn git_head(&self) -> Option<String>;
+
+    /// return the date of the last commit of the repository.
+    fn date_of_the_last_commit(&self) -> Option<String>;
```

Comment:
This is a bad design, you should remove the method for the `Repository` Trait. It is an implementation-specific information, so this should stay in `coffee_github`

Reply:
This is just the definition of these methods
The actual implementation is in `coffee_github/src/repository.rs`
We have to implement these methods here to be able to call them on variables on type `Repository` in `coffee_core/src/coffee.rs` 
For example in `list_remotes` method

Reply:
>This is just the definition of these methods

This is the problem. You require this method in all the implementations of this trait, and this is wrong. You should implement this only inside `coffee_github`.

>We have to implement these methods here to be able to call them on variables on type Repository in coffee_core/src/coffee.rs
For example, in list_remotes method

Why not call just `get_info` in these places?

Reply:
> Why not call just get_info in these places?
We definitely can

Reply:
yeah so we should only expose the get_info and then implement in the `coffee_github` the others two methods